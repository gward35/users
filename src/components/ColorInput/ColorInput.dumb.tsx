import {
  Box,
  Flex,
  InputLeftAddon,
  InputGroup,
  Input,
  Text,
  Tooltip,
} from '@chakra-ui/react'
import { InfoOutlineIcon } from '@chakra-ui/icons'
import { ChakraProps } from '@chakra-ui/react'

export interface DumbColorInputProps extends ChakraProps {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  isDisabled: boolean
  isLargerThan768: boolean
  isValid: boolean
  value: string
}

export const DumbColorInput = ({
  onChange,
  isDisabled,
  isLargerThan768,
  isValid,
  value,
}: DumbColorInputProps) => {
  return (
    <Flex width="100%" justifyContent="center" px={isLargerThan768 ? 0 : 4}>
      <Flex flexDirection="column" width={isLargerThan768 ? '50%' : '100%'}>
        <Box as="label" htmlFor="changeColor" hidden />
        <InputGroup>
          <InputLeftAddon>
            <Tooltip
              label={
                'Valid HEX format starts with a # and can be three or six characters long e.g. #eee or #ff0000'
              }
              aria-label="tooltip"
            >
              <InfoOutlineIcon data-testid="infoIcon" color="primary" />
            </Tooltip>
          </InputLeftAddon>
          <Input
            borderColor={!isValid && value && !isDisabled ? '#ff0000' : 'inherit'}
            isDisabled={isDisabled}
            id="changeColor"
            data-testid="colorInput"
            name="color"
            placeholder={value ? value : 'Change color'}
            onChange={onChange}
            type="text"
          />
        </InputGroup>
        {!isValid && value && !isDisabled && (
          <Text
            fontSize="sm"
            data-testid="errorMsg"
            display="block"
            color="danger.100"
          >
            Input is not valid. Please try again.
          </Text>
        )}
      </Flex>
    </Flex>
  )
}

export default DumbColorInput
