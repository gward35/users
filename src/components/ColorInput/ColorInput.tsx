import DumbColorInput from './ColorInput.dumb'

export type ColorInputProps = {
  isDisabled: boolean
  isLargerThan768: boolean
  isValid: boolean
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  value: string
}

export const ColorInput = ({
  isDisabled,
  isLargerThan768,
  isValid,
  onChange,
  value,
}: ColorInputProps) => {
  return (
    <DumbColorInput
      onChange={onChange}
      isDisabled={isDisabled}
      isLargerThan768={isLargerThan768}
      isValid={isValid}
      value={value}
    />
  )
}

export default ColorInput
