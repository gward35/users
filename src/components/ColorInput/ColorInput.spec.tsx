import { render } from '@testing-library/react'
import ColorInput from './ColorInput'

describe('Color Input', () => {
  const PROPS = {
    isDisabled: false,
    isValid: true,
    onChange: jest.fn(),
    value: '',
  }

  it('renders component and snaps', () => {
    const utils = render(<ColorInput {...PROPS} />)
    expect(utils.container).toMatchSnapshot()
  })
})
