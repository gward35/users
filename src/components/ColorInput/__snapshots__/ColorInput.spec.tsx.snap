// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Input renders component and snaps 1`] = `
<div>
  <div
    class="css-1h1xr4x"
  >
    <div
      class="css-1ofqig9"
    >
      <label
        class="css-0"
        for="changeColor"
        hidden=""
      />
      <div
        class="chakra-input__group css-bx0blc"
        data-group="true"
      >
        <div
          class="chakra-input__left-addon css-1lhl5nw"
        >
          <svg
            class="chakra-icon css-1fnxnin"
            data-testid="infoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              fill="currentColor"
              stroke="currentColor"
              stroke-linecap="square"
              stroke-width="2"
            >
              <circle
                cx="12"
                cy="12"
                fill="none"
                r="11"
                stroke="currentColor"
              />
              <line
                fill="none"
                x1="11.959"
                x2="11.959"
                y1="11"
                y2="17"
              />
              <circle
                cx="11.959"
                cy="7"
                r="1"
                stroke="none"
              />
            </g>
          </svg>
        </div>
        <input
          class="chakra-input css-1m5purn"
          data-testid="colorInput"
          id="changeColor"
          name="color"
          placeholder="Change color"
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
